### DISCUSSION TITLE
Installed into Jupyterlab but no icon appearing in left side menu

### DISCUSSION THREAD

**Discussion by aurorasea123** [jupyterlab/jupyterlab-git#983] [Q&A]
Hello I am using version 3.0.14 of Jupyterlab.
I managed to install this using extension manager, which was successful but the icon failed to appear in the left menu bar. 
Thank you for helping with this.

**Comment by krassowski** [comment]
Hi @aurorasea123 thanks for posting the question. There are two things to mention:

1. installation from extension manager is not sufficient as there is also a server extension that needs to be installed and currently the extension manager does not handle that yet. I personally would uninstall the version installed with the extension manager and instead follow the instructions in the readme file this is:

    ```
    pip install --upgrade jupyterlab jupyterlab-git
    # or, if using conda:
    # conda install -c conda-forge jupyterlab jupyterlab-git
    ```

2. a restart of JupyterLab will be needed after the installation is complete; while the icon may appear just after refresh, you will still need restart JupyterLab to get the server extension kick-in.

If after following steps (1) and (2) you still cannot get it working, please check what is your:

```
jupyter labextension list
```

and:

```
jupyter server extension list
```

**Reply by aurorasea123** [reply]
Uninstalled

Re-installed git as instructed, restarted

Error message appeared: git command not found - please ensure you have git > 2 installed. I still cannot see extension icon

jupyter labextension list
JupyterLab v3.0.16
d:\user\localdocuments\bioinformatics\jupyterlab\share\jupyter\labextensions
nbdime-jupyterlab v2.1.0 enabled ok
@jupyterlab/git v0.31.0 enabled ok (python, jupyterlab-git)

jupyter server extension list
Config dir: C:\Users\auror.jupyter

Config dir: D:\User\LocalDocuments\Bioinformatics\Jupyterlab\etc\jupyter
jupyter_server_mathjax enabled
- Validating jupyter_server_mathjax...
jupyter_server_mathjax ok
jupyterlab enabled
- Validating jupyterlab...
jupyterlab 3.0.16 ok
jupyterlab_git enabled
- Validating jupyterlab_git...
jupyterlab_git 0.31.0 ok
nbclassic enabled
- Validating nbclassic...
nbclassic ok
nbdime enabled
- Validating nbdime...
nbdime 3.1.0 ok

Config dir: C:\ProgramData\jupyter

It is still not working and I am not sure what has gone wrong.
Thank you

**Reply by fcollonval** [reply]
Hey

Could you check you have git available?

How to
1. Open a console within JupyterLab
   1. Menu _File_ -> _New_ -> _Console_
2. Execute `!git --version`

It should print a version if git is installed. If not, you can install it using conda (`conda install git`) or from the [Git website](https://git-scm.com/)

**Reply by aurorasea123** [reply]
I checked and installed : it is git version 2.23.0.windows.1

Somehow the menu did not appear. 
I reinstalled nodejs in the console and now the icon appears

**Reply by fcollonval** [reply]
Did you uninstall the extension from the extension manager before installing it with `pip` (or `conda`)?

If not this can explain why you had to install nodejs. You can avoid nodejs by running `pip install jupyterlab jupyterlab-git` in any environment without using the extension manager.
